src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,17 +30,98 @@
+    public static boolean isValidAddress(String address) throws IllegalValueException{
+        try {
+            HashMap<String, String> addressParams = extractAddressParam(address);
+            String block = addressParams.get("block");
+            String street = addressParams.get("street");
+            String unit = addressParams.get("unit");
+            String postalCode = addressParams.get("postalCode");
+            
+            if (!Block.isValidBlock(block)) {
+                throw new IllegalValueException(Block.MESSAGE_BLOCK_CONSTRAINTS);
+            } else if (!Street.isValidStreet(street)) {
+                throw new IllegalValueException(Street.MESSAGE_STREET_CONSTRAINTS);
+            } else if (!Unit.isValidUnit(unit)) {
+                throw new IllegalValueException(Unit.MESSAGE_UNIT_CONSTRAINTS);
+            } else if (!PostalCode.isValidPostalCode(postalCode)) {
+                throw new IllegalValueException(PostalCode.MESSAGE_POSTAL_CODE_CONSTRAINTS);
+            }
+        } catch (IllegalValueException e) {
+            throw e;
+        }
+        return true;
+    }
+    
+    /**
+     * Extract the different parts of the address
+     * 
+     * @return HashMap containing the different parts of the address
+     */
+    private static HashMap<String, String> extractAddressParam(String address)throws IllegalValueException {
+        StringTokenizer st = new StringTokenizer(address, ",");
+        String[] addressParamArr = new String[ADDRESS_PARAM_COUNT];
+        int count=0;
+        while (st.hasMoreTokens()) {
+            addressParamArr[count++] = st.nextToken().trim();
+        }
+        
+        if (count == ADDRESS_PARAM_COUNT) {
+            HashMap<String, String> addressParamHashMap = new HashMap<String, String>();
+            addressParamHashMap.put("block", addressParamArr[0]);
+            addressParamHashMap.put("street", addressParamArr[1]);
+            addressParamHashMap.put("unit", addressParamArr[2]);
+            addressParamHashMap.put("postalCode", addressParamArr[3]);
+            return addressParamHashMap;
+        }
+        else {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+    }
+    
+    /**
+     * Update the block, street, unit, postalCode members of Address
+     * @param address
+     * @throws IllegalValueException
+     */
+    private void partitionAddress(String address) throws IllegalValueException{
+        try { 
+            HashMap<String, String> addressParams = extractAddressParam(address);
+            block = new Block(addressParams.get("block"));
+            street = new Street(addressParams.get("street"));
+            unit = new Unit(addressParams.get("unit"));
+            postalCode = new PostalCode(addressParams.get("postalCode"));
+        } catch (IllegalValueException e) {
+            throw e;
+        }
+    }
+    
+    public Block getBlock() {
```

Comment:
why would you need these objects directly? you can have/use accessors in the respective class to update (if it needs to be done) a particular field, no?